src/cpp/session/resources/grid/datatables/js/jquery.dataTables.js

```diff
@@ -927,18 +927,7 @@
 	
 				for ( j=0, jLen=aTargets.length ; j<jLen ; j++ )
 				{
-					if ( typeof aTargets[j] === 'number' && aTargets[j] >= 0 )
-					{
-						/* Add columns that we don't yet know about */
-						while( columns.length <= aTargets[j] )
-						{
-							_fnAddColumn( oSettings );
-						}
-	
-						/* Integer, basic index */
-						fn( aTargets[j], def );
-					}
-					else if ( typeof aTargets[j] === 'number' && aTargets[j] < 0 )
+					if ( typeof aTargets[j] === 'number' && aTargets[j] < 0 )
```

Comment:
Do we have a mechanism for preserving these patches if we update jQuery in the future?

Reply:
I'm not sure, though this is an iframe'd in javascript that lives in its own sandbox and the versions of jQuery and DataTables in here are quite old. I'm of the mind right now to push for a rewrite of the DataTable feature in general when it is able to get priority at this point regardless.

Reply:
I think that when/if we rewrite the data viewer we should use another library entirely. DataTables is not really up to the task. PhosphorJS for example has a datagrid library that purportedly can handle arbitrarily large data (they have a demo that does 1 trillion rows/cols smoothly).

https://github.com/phosphorjs/phosphor/tree/master/packages/datagrid

Reply:
Absolutely. DataTables/jQuery are very, very old tech at this point make things way harder than they have to be. A Flexbox-based styling would literally cut DataTable's code in half, or less. There's a lot of stuff that it does that is just antiquated these days.

Reply:
Sounds good -- if we're anticipating an infrastructure overhaul next time we spend substantial time on the viewer then I think it'd fine to just patch directly.